clients/src/test/java/org/apache/kafka/clients/NetworkClientTest.java

```diff
@@ -408,9 +408,6 @@ public void testUnsupportedApiVersionsRequestWithVersionProvidedByTheBroker() {
         // ApiVersionsRequest is in flight but not sent yet
         assertTrue(client.hasInFlightRequests(node.idString()));
 
-        // ApiVersionsResponse has been received
-        assertEquals(1, selector.completedReceives().size());
```

Comment:
why this assert was removed? what's the size of completedReceives?

Reply:
I don't think this assert is really necessary. All this assert tries to verify is that the Selector received a response. Though that is already verified because by making sure that client has in flight request inside. 

Another option to really verify this is via a counter based mechanism in MockSelector and using the count to verify there was a response that was received. Let me know if that sounds better?